packages/authentication/test/unit/authentication.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from 'testlab';
+import {Authentication} from '../../src/authentication';
+import {BasicStrategy} from 'passport-http';
+const http = require('http');
+
+describe('Authentication', () => {
+  let passport;
+  let strategy;
+  let req;
+  let res;
+
+  describe('Basic Strategy', () => {
+    beforeEach(createBasicStrategy);
+    beforeEach(createPassport);
+    beforeEach(createMockHttpObjects);
+
+    it('authenticates successfully for correct credentials', () => {
+      const credential = 'simpson' + ':' + 'alpha';
+      const hash = new Buffer(credential).toString('base64');
+      req.headers.authorization = 'Basic ' + hash;
+      passport.authenticate(req, res, function done(err) {
+        expect(err).to.be.undefined;
+        expect(req.user.role).to.equal('admin');
```

Comment:
When `req.user` is not set, this assertion reports a TypeError before `expect` is even called:

```ts
expect(req.user.role).to.equal('admin');
// TypeError: Cannot read property 'role' of undefined
```

A better way is to use `.have.property` checker:

```ts
expect(req.user).to.have.property('role', 'admin');
// AssertionError: expected undefined to have property role
```